### PR TITLE
[Analyzer] Warn when integrating dynamic frameworks into CLI targets

### LINKED ISSUE #6493: The dyld: Library not loaded issue
# Report

## What did you do?

When running a clean, empty, Swift-based "hello world" command-line project in XCode  8.3 beta 2 (8W120l) or AppCode 2016.3.2 and having a minimal Podfile with (seemingly) any Pod, the project builds, but run leads to errors such as (for example)

`dyld: Library not loaded: @rpath/Alamofire.framework/Versions/A/Alamofire
  Referenced from: /Users/julian/Library/Developer/Xcode/DerivedData/wtf-cyoeewosqluzxmcwypdhyqrdoqlq/Build/Products/Debug/wtf
  Reason: image not found`

The project makes no direct or indirect reference to code in the Pod (i.e., no import and no invocations.) Running the executable from within either IDE or at the command line leads to the same error.

## What did you expect to happen?
The program to print "hello world"

## What happened instead?
Program ended with error 9

### Stack

```
   CocoaPods : 1.2.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.4 (16E144f)
       Xcode : 8.2.1 (8C1002)
         Git : git version 2.11.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 0e21f945f392282fa53bdc92bcecc7f9bbeca830
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate     : 1.0.1
cocoapods-playgrounds     : 1.2.2
cocoapods-plugins         : 1.0.0
cocoapods-search          : 1.0.0
cocoapods-stats           : 1.0.0
cocoapods-trunk           :
```

## Project that demonstrates the issue

https://www.dropbox.com/s/fx58gv1n94qjyoq/wtf.tgz?dl=0

### PR DESCRIPTION
**Pull request by amorde**
Closes #6493

### CODE REVIEW

File: spec/unit/installer/analyzer_spec.rb [hunk 1]
```diff
@@ -1911,19 +1911,20 @@ module Pod
           UI.warnings.should.match /The Podfile contains framework or static library targets \(SampleLib\), for which the Podfile does not contain host targets \(targets which embed the framework\)\./
         end
 
-        it 'warns when using a Podfile for framework-only projects' do
+        it 'warns when using dynamic frameworks with CLI targets' do
```
Comment (amorde) [inline spec/unit/installer/analyzer_spec.rb]: diff looks weird, but the existing test was a dupe of the one directly above it

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -354,6 +358,12 @@ def analyze_host_targets_in_podfile(aggregate_targets, embedded_aggregate_target
           end
         end
 
+        unless cli_host_with_frameworks.empty?
+          UI.warn "The Podfile contains command line tool target(s) (#{cli_host_with_frameworks.map(&:name).to_sentence}) which are attempting to integrate dynamic frameworks." \
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer.rb]: nit wrap the target names with  backticks

**Comment by dnkoutso** [comment]
just rubocop

File: lib/cocoapods/installer/analyzer.rb [hunk 3]
```diff
@@ -324,21 +324,25 @@ def embedded_target_pod_targets_by_host(aggregate_target, embedded_aggregate_tar
...
       def analyze_host_targets_in_podfile(aggregate_targets, embedded_aggregate_targets)
         target_definitions_by_uuid = {}
+        cli_host_with_frameworks = []
+        cli_product_type = 'com.apple.product-type.tool'
         # Collect aggregate target definitions by uuid to later lookup host target
         # definitions and verify their compatiblity with their embedded targets
         aggregate_targets.each do |target|
-          target.user_targets.map(&:uuid).each do |uuid|
-            target_definitions_by_uuid[uuid] = target.target_definition
+          target.user_targets.each do |user_target|
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer.rb]: can we not do `cli_host_with_frameworks = target.user_targets.map(&:uuid).select do ... end`?

Comment (dnkoutso) [reply lib/cocoapods/installer/analyzer.rb]: ah nm I see this does two things

File: lib/cocoapods/installer/analyzer.rb (hunk 2 above)
Comment (amorde) [reply lib/cocoapods/installer/analyzer.rb]: kept it as this style to match https://github.com/CocoaPods/CocoaPods/pull/8767/files#diff-f5c3fee776e864ada6541431a1532b00R376

**Pull request merged**